// edamam-node/lib/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DietLabel;
(function (DietLabel) {
    DietLabel["Balanced"] = "balanced";
    DietLabel["HighFiber"] = "high-fiber";
    DietLabel["HighProtein"] = "high-protein";
    DietLabel["LowCarb"] = "low-carb";
    DietLabel["LowFat"] = "low-fat";
    DietLabel["LowSodium"] = "low-sodium";
})(DietLabel = exports.DietLabel || (exports.DietLabel = {}));
var HealthLabel;
(function (HealthLabel) {
    HealthLabel["AlcoholFree"] = "alcohol-free";
    HealthLabel["CeleryFree"] = "celery-free";
    HealthLabel["CrustaceanFree"] = "crustacean-free";
    HealthLabel["Dairy"] = "dairy-free";
    HealthLabel["Eggs"] = "egg-free";
    HealthLabel["Fish"] = "fish-free";
    HealthLabel["Gluten"] = "gluten-free";
    HealthLabel["KidneyFriendly"] = "kidney-friendly";
    HealthLabel["Kosher"] = "kosher";
    HealthLabel["LowPotassium"] = "low-potassium";
    HealthLabel["LupineFree"] = "lupine-free";
    HealthLabel["MustardFree"] = "mustard-free";
    HealthLabel["NoOilAdded"] = "No-oil-added";
    HealthLabel["NoSugar"] = "low-sugar";
    HealthLabel["Paleo"] = "paleo";
    HealthLabel["Peanuts"] = "peanut-free";
    HealthLabel["Pescatarian"] = "pescatarian";
    HealthLabel["PorkFree"] = "pork-free";
    HealthLabel["RedMeatFree"] = "red-meat-free";
    HealthLabel["SesameFree"] = "sesame-free";
    HealthLabel["Shellfish"] = "shellfish-free";
    HealthLabel["Soy"] = "soy-free";
    HealthLabel["SugarConscious"] = "sugar-conscious";
    HealthLabel["TreeNuts"] = "tree-nut-free";
    HealthLabel["Vegan"] = "vegan";
    HealthLabel["Vegetarian"] = "vegetarian";
    HealthLabel["WheatFree"] = "wheat-free";
})(HealthLabel = exports.HealthLabel || (exports.HealthLabel = {}));
var NutrientCode;
(function (NutrientCode) {
    NutrientCode["Calcium"] = "CA";
    NutrientCode["Carbs"] = "CHOCDF";
    NutrientCode["Cholesterol"] = "CHOLE";
    NutrientCode["Monounsaturated"] = "FAMS";
    NutrientCode["Polyunsaturated"] = "FAPU";
    NutrientCode["Saturated"] = "FASAT";
    NutrientCode["Fat"] = "FAT";
    NutrientCode["Trans"] = "FATRN";
    NutrientCode["Iron"] = "FE";
    NutrientCode["Fiber"] = "FIBTG";
    NutrientCode["Folate"] = "FOLDFE";
    NutrientCode["Potassium"] = "K";
    NutrientCode["Magnesium"] = "MG";
    NutrientCode["Sodium"] = "NA";
    NutrientCode["Energy"] = "ENERC_KCAL";
    NutrientCode["Niacin"] = "NIA";
    NutrientCode["Phosphorus"] = "P";
    NutrientCode["Protein"] = "PROCNT";
    NutrientCode["Riboflavin"] = "RIBF";
    NutrientCode["Sugars"] = "SUGAR";
    NutrientCode["Thiamin"] = "THIA";
    NutrientCode["VitaminE"] = "TOCPHA";
    NutrientCode["VitaminA"] = "VITA_RAE";
    NutrientCode["VitaminB12"] = "VITB12";
    NutrientCode["VitaminB6"] = "VITB6A";
    NutrientCode["VitaminC"] = "VITC";
    NutrientCode["VitaminD"] = "VITD";
    NutrientCode["VitaminK"] = "VITK1";
})(NutrientCode = exports.NutrientCode || (exports.NutrientCode = {}));
